### ISSUE TITLE
Improvements and patches required on LECO

### ISSUE THREAD

**Issue by rgeneaux** [PyMoDAQ/PyMoDAQ#382]
Edit: while working on LECO communication, a few issues were identified:

- LECO communication prints unwanted messages
- The communication on remote IP adresses is not implemented at the moment
- LECO does not handle actuators with units (added in 4.3.0)

Original message:
---------------------------
Hi,
I've started using the LECO communication to control daq_moves from our ARPES software. The software instantiates a DAQ_Move_LECODirector, and then uses stdin/stdout to control it. It really works great in general (thanks @BenediktBurger), it is really resilient.
One last issue I have is that whenever commands are sent, a number of messages are printed in the console, e.g.:
```
Threadcommand: units with attribute 
Threadcommand: update_settings with attribute [['bounds'], {'visible': False}, 'options']
Threadcommand: update_settings with attribute [['scaling'], {'visible': False}, 'options']
Threadcommand: update_settings with attribute [['actor_name'], 'Delay', 'value']
Threadcommand: update_settings with attribute [['units'], {'visible': False}, 'options']
Threadcommand: update_settings with attribute [['epsilon'], {'visible': False}, 'options']
Threadcommand: get_actuator_value with attribute [<DataActuator (5.000374093849458)>]
```
This is great for debugging, but actually interferes with the communication based on stdin. I can filter some of them out, but not all (since they might arrive asynchronously).
Is there a way to disable all these to be printed? I wasn't able to figure out where they come from.

Thanks!

**Comment by seb5g** [comment]
I think Benedikt is the one who could quickly tell!

**Comment by BenediktBurger** [comment]
I'm not entirely sure, as I do not have a total oversight of PyMoDAQs logging/printing.

The printing line might be here:
https://github.com/PyMoDAQ/PyMoDAQ/blob/0380583dd08bc306b630bab930ebfab07fa3041a/src/pymodaq/control_modules/move_utility_classes.py#L460-L467

Any command will also emit a status, for example: https://github.com/PyMoDAQ/PyMoDAQ/blob/0380583dd08bc306b630bab930ebfab07fa3041a/src/pymodaq/utils/leco/daq_move_LECODirector.py#L156-L158

That status will propagate and will be printed.

I had added these logging calls, as I thought them useful (maybe they are also in the normal classes, I do not recall), without fully understand how they are handled later on.

Right now, I cannot test it. Later I can take another look, if you want to.

**Comment by seb5g** [comment]
It's weird they are printed on the sdtout, as messages or status should be
added in the log...

Le mer. 25 sept. 2024 à 18:56, Benedikt Burger ***@***.***> a
écrit :

**Comment by BenediktBurger** [comment]
Could it be, that the remote window does not have a parent as it is not part of the dashboard, therefore it prints the status instead of logging them?

**Comment by rgeneaux** [comment]
Exactly, I just checked and removing line 467 removes all of the messages. I guess we could replace that print line by writing to the log instead? DEBUG or INFO level?

**Comment by seb5g** [comment]
I guess you could replace the print line by:

```
logger.debug(whatever message should be logged)
```
as debug because it is not really interesting for other purpose!

**Comment by rgeneaux** [comment]
Hi, small update on this topic:

- Indeed moving the messages to debug fixed my issues
- When trying to use LECO across a network, I realized that in the current implementation it only works on localhost (ip adrresses are never passed to the listener so it just uses default). I did a small fix on this, which works (for our experiment things work great)
- I wanted to do a PR with including these fixes, so I switched to the latest 4.X version, but now with the inclusion of units, things don't work anymore, the Director receives messages that it is not able to understand. Also the DAQ_Move_Director needs to be updated to include units.
- I hope I manage to debug this and eventually do a PR!

So this issue remains on hold for now. I rename the issue to include other things

**Comment by seb5g** [comment]
Alright,
I'm also working on a PR to better deal with units, epilsons in actuator plugins depending if the axes have bben defined as list or dict.

**Comment by seb5g** [comment]
I think recent PR solved all this

#554
